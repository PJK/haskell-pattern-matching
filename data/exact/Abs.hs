module Abs where

abs :: Int -> Int
abs x
    | x < 0 = - x
    | x > 0 = x