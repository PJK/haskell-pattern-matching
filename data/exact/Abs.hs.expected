{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "abs",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "VariablePattern",
                                    "contents": "~a"
                                }
                            ],
                            "Constraints:\n  ~f == False\n  ~f == ~a > 0\n  ~c == False\n  ~c == ~a < 0\n\nSatisfiable. Model:\n  ~f = False :: Bool\n  ~a =     0 :: Integer\n  ~c = False :: Bool\n"
                        ]
                    ]
                }
            ]
        ],
        [
            [
                "abs",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
